contracts/oracle/BackedOracle.sol

```diff
@@ -0,0 +1,199 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2021-2023 Backed Finance AG
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+pragma solidity 0.8.9;
+
+import "./BackedOracleInterface.sol";
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract BackedOracle is OwnableUpgradeable, AggregatorV2V3Interface {
+    uint8 public constant VERSION = 1;
+    uint8 public constant MAX_PERCENT_DIFFERENCE = 10;
+    uint32 public constant MAX_TIMESTAMP_AGE = 5 minutes;
+    uint32 public constant MIN_UPDATE_INTERVAL = 1 hours;
+
+    struct RoundData {
+        int192 answer;
+        uint32 timestamp;
+    }
+
+    uint8 private _decimals;
+    string private _description;
+
+    mapping(uint256 => RoundData) private _roundData;
+    uint80 private _latestRoundNumber;
+
+    constructor() {
+        initialize(8, "Backed Oracle");
```

Comment:
This is not wrong, but just to be sure:  given that this contract will be used as the logic implementation used by proxies, it does not make much sense to configure it with decimals and a description.

Reply:
you could set the decimals to 0 and change the name t "Backed Oracle Implemention" so there is no ambiguity